File: spa/src/types/reservation.ts
import type { ReservationStatus } from './ReservationStatus';
import type { Table } from './Table';
import type { TableData } from './TableData';

export interface Reservation extends TableData {
  [key: string]: string | number | Table | null;
  id: number;
  name: string;
  day: string;
  hour: string;
  people: number;
  table: Table | null;
  notes: string | null;
  status: ReservationStatus;
}

File: MinimalAPI/Data/Reservation.cs
using MinimalAPI.Common;

namespace MinimalAPI.Data
{
    public class Reservation : IEntity, ICanBeDeleted
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateOnly Day { get; set; }
        public TimeOnly Hour { get; set; }
        public int People { get; set; }
        public Table? Table { get; set; }
        public string? Notes { get; set; }
        public ReservationStatus Status { get; set; }
        public bool IsDeleted { get; set; }
    }
}
